src/compas/geometry/primitives/polyline.py

```diff
@@ -412,4 +412,74 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
 
         return division_pts
 
+    def extend(self, length, both_ends=False):
+        """Extends a polyline from its start or end by a given length
+
+        Parameters:
+        -----------
+        length: list of float [start_offset, end_offset]
```

Comment:
We now write these a bit different, in this case, I think it would be:
```suggestion
        length: list[float, float]
```
@tomvanmele is that correct?

Reply:
not really.
you can't specify a fixed length sequence as a list.
you could do `tuple[float, float]`, but we have been using `[float, float]` in case any sequence of 2 floats is acceptable...